src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt

```diff
@@ -178,9 +181,11 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
                 ?:
             throw IllegalArgumentException("No replication in progress for index:${request.indexName}")
         val replicationOverallState = replicationStateParams[REPLICATION_LAST_KNOWN_OVERALL_STATE]
-        if (replicationOverallState == ReplicationOverallState.RUNNING.name)
+        if (replicationOverallState == ReplicationOverallState.PAUSED.name)
```

Comment:
Based on previous logic, this should return true in "Pause" state?

Reply:
Yeah, just realized and updated.